### DISCUSSION TITLE
How to handle 1000000 clients?

### DISCUSSION THREAD

**Discussion by rajcse2005** [moscajs/aedes#791] [General]
Any one let me know maximum subscriber can connect or unlimited?
Which parameters can control it

**Comment by robertsLando** [comment]
There is no real limit on code side, it depends on the machine that is running aedes. Worth mentioning that `concurrency` option must be set to 0 in that case or a very high value

**Reply by rajcse2005** [reply]
Thanks for your reply.
If I want handle that much subscriber. What is perfect machine configuration?

**Reply by Gianluca-Casagrande-Stiga** [reply]
Hi, there are many variables, first of all messages frequency and payload size. Are they constant o they could fluctuate? I think you should do some empirical attempts to get the best server setup.

**Reply by rajcse2005** [reply]
Is there any payload size limitation?
My case max will be less then100kb.

**Comment by rajcse2005** [comment]
Is there any payload size limitation?
My case max will be less then100kb.

**Reply by Gianluca-Casagrande-Stiga** [reply]
The max MQTT payload size should be 256MB, I've done some tests and I've found some problem over the 80MB, but I'm not sure that it was due to Aedes.

**Reply by rajcse2005** [reply]
I have tested with 15ksubscriber it running perfectly. If add more subscribers then it create an issue. I am not sure it is because of size issue or anything else.
